src/indieweb_utils/feeds/discovery.py

```diff
@@ -47,12 +52,16 @@ def discover_web_page_feeds(url: str, user_mime_types: Optional[List[str]] = Non
         url = "https://" + url
     elif url.startswith("//"):
         url = "https:" + url
-    try:
-        web_page_request = requests.get(url, timeout=10, allow_redirects=True)
 
+    if web_page_request:
         web_page = web_page_request.text
-    except requests.exceptions.RequestException:
-        return []
+    else:
+        try:
+            web_page_request = requests.get(url, timeout=10, allow_redirects=True)
+        except requests.RequestException:
+            raise Exception("Request to retrieve URL did not return a valid response.")
+
+    web_page = web_page_request.text
```

Comment:
Assuming we rename `web_page_request` to `html` and it's now a string as suggested in my previous comment - I think we could make this a bit more concise.

```py
if not html:
    try:
         web_page_request = requests.get(url, timeout=10, allow_redirects=True)
    except requests.RequestException:
        # It's not clear why we're re-raising this exception as a generic exception. A comment would be helpful.
        raise Exception("Request to retrieve URL did not return a valid response.")
    else:
        html = web_page_request.text
```